sourmash_lib/search.py

```diff
@@ -162,13 +163,16 @@ def build_new_signature(mins, template_sig):
         R_genome = best_leaf.minhash.scaled
 
         # pick the highest R / lowest resolution
-        R_comparison = max(R_metagenome, R_genome)
+        R_comparison = max(R_comparison, R_metagenome, R_genome)
 
-        # CTB: these could probably be replaced by minhash.downsample_scaled.
-        new_max_hash = sourmash_lib.MAX_HASH / float(R_comparison)
+        # eliminate mins under this new resolution.
+        # (CTB note: this means that if a high scaled/low res signature is
+        # found early on, resolution will be low from then on.)
+        new_max_hash = get_max_hash_for_scaled(R_comparison)
         query_mins = set([ i for i in query_mins if i < new_max_hash ])
         found_mins = set([ i for i in found_mins if i < new_max_hash ])
         orig_mins = set([ i for i in orig_mins if i < new_max_hash ])
+        sum_abunds = sum([ v for (k,v) in orig_abunds.items() if k < new_max_hash ])
```

Comment:
I don't understand why not having this in previously didn't cause a problem, but I understand why it should be present